systemd: Simplify unit script names

kirelagin opened the pull request:
Make journalctl logs readable again!

* Remove the superfluous `unit-script` prefix. Personally I can tell it’s a script by its distinctive `-start` / `-preStart` / etc. suffix.
* Put it back into `/bin/` (https://github.com/NixOS/nixpkgs/commit/adba92b5ef37e2fb873c29a07d7760119c8701b7 😒) because I don’t want to see those hashes in my `journalctl` output (this and some services printing their own timestamps makes the actual log message start on 3/4 of my screen, and I am not exaggerating).

cc @lheckemann

kirelagin committed:
systemd: Simplify unit script names

Current journal output from services started by `script` rather than
`ExexStart` is unreadable because the name of the file (which journalctl
records and outputs) quite literally takes 1/3 of the screen (on smaller
screens).

Make it shorter. In particular:

* Drop the `unit-script` prefix as it is not very useful.
* Use `writeShellScriptBin` to write them because:
  * It has a `checkPhase` which is better than no checkPhase.
  * The script itself ends up having a short name.

kirelagin commented:
Oh, I just noticed that @lheckemann added this `unix-script` prefix in the same commit. Are we going to fight? 🤔
I suppose, I can concede either `unit-` or `script-`, but not both. (And to be honest, I still don’t see the need in either of those.)

lheckemann commented:
see discussion at https://github.com/NixOS/nixpkgs/pull/43534

zhenyavinogradov commented on nixos/modules/system/boot/systemd.nix:
```diff
@@ -201,8 +201,13 @@ let
-    in  pkgs.writeTextFile { name = mkScriptName name; executable = true; inherit text; };
+    let
+      scriptName = replaceChars [ "\\" "@" ] [ "-" "_" ] (shellEscape name);
+      out = pkgs.writeShellScriptBin scriptName ''
```
`writeShellScriptBin` will include the shebang, so you can remove it from `makeJobScript` invocations below

kirelagin replied:
Oh, yes, I meant to remove it, thanks.

kirelagin committed:
fixup! systemd: Simplify unit script names

kirelagin commented:
Looking at the discussion, it seems that people just like seeing `unit-script` in their `/nix/store` paths, is that right? In this case a solution I see is to use `writeTextFile` and add `unit-script` in the derivation name, but not in the script file name.

(However, personally I would prefer not to do this to keep things simple.)

cc @fpletz @globin @grahamc

kirelagin committed:
systemd: Add prefix to unit script derivations

Add a distinctive `unit-script` prefix to systemd unit scripts to make
them easier to find in the store directory. Do not add this prefix to
actual script file name as it clutters logs.

kirelagin commented:
If people insist on keeping the prefix, here is a commit that implements this ^. However, as I said, I would really prefer to drop it.

kirelagin committed:
fixup! systemd: Add prefix to unit script derivations

* Avoid extra string interpolation.

Mic92 commented on nixos/modules/system/boot/systemd.nix:
```diff
@@ -201,8 +201,23 @@ let
+          ${text}
+        '';
+        checkPhase = ''
+          ${pkgs.stdenv.shell} -n "$out/bin/${scriptName}"
```
I like this part.

Mic92 replied:
Seems to be the default of `writeShellscriptBin` anyway.

kirelagin replied:
Yes, I copied it from there.

kirelagin commented:
ping

kirelagin commented:
👋

lheckemann commented:
@ofborg test boot

lheckemann commented:
I'd merge this if the tests pass and nobody voices any objections until then.

bjornfor commented:
Thanks for fixing this!

kirelagin commented:
Thanks! If you liked this change, please, help me get NixOS/nixpkgs#87219 merged too :).

datafoo commented:
Will these changes be back-ported to `nixos-20.03-small` and `nixos-20.03` channels?

datafoo commented:
Sorry if that sounds stupid. I haven't fully understand how/when changes committed on the master branch of this repo end up in a channel.
I was interested to know if these changes will be visible here: https://github.com/NixOS/nixpkgs-channels/blob/nixos-20.03-small/nixos/modules/system/boot/systemd.nix

Mic92 commented:
We usually backport bugfixes and keep major changes in master. I don't know in which category this change falls. If the maintainer don't want to backport it, it will be available in 20.09.
